services/plugins/generic_plugins/schemas.go

```diff
@@ -0,0 +1,135 @@
+package generic_plugins
+
+import (
+	"github.com/leanspace/terraform-provider-leanspace/helper"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+var validGenericPluginTypes = []string{
+	"CHECKSUM_FUNCTION",
+}
+
+var genericPluginSchema = map[string]*schema.Schema{
+	"id": {
+		Type:     schema.TypeString,
+		Computed: true,
+	},
+	"name": {
+		Type:     schema.TypeString,
+		Required: true,
+	},
+	"description": {
+		Type:     schema.TypeString,
+		Optional: true,
+	},
+	"type": {
+		Type:         schema.TypeString,
+		Required:     true,
+		ValidateFunc: validation.StringInSlice(validGenericPluginTypes, false),
+		Description:  helper.AllowedValuesToDescription(validGenericPluginTypes),
+	},
+	"language": {
+		Type:     schema.TypeString,
+		Required: true,
```

Comment:
Can you also provide the ValidateFunc, there's only JAVA allowed as far as I know

Reply:
Ok done. For the moment, yes it's just JAVA but it should evolve to other language.